src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -62,6 +62,9 @@ public sealed partial class FileSystemProvider : NavigationCmdletProvider,
 
         private const int COPY_FILE_ACTIVITY_ID = 0;
         private const int REMOVE_FILE_ACTIVITY_ID = 0;
+        
+        // Windows error code for invalid characters
+        private const int ERROR_INVALID_NAME = unchecked((int)0x8007007B);
```

Comment:
What about Linux and MacOS?

Reply:
Unfortunately, I have limited knowledge about Linux or macOS. From what I have read, the only invalid character for directory name on Linux is `/` (forward slash). I did try force an exception on Ubuntu with `New-Item -Path Invalid/folder/name' -Type Directory`, and it does not error but instead creates a folder structure. What I can do is not suppress `IOException` when on Windows.

Reply:
@mklement0 What do you think about non-Windows OS-s? Do you agree that the fix should be only for Windows?

Reply:
Since we don't use `RuntimeInformation.IsOSPlatform()` in the file, we prefer to wrap code with `#if UNIX`. So I suggest to move the const to code where it is used and use the directive.